fgspectra/frequency.py

```diff
@@ -3,120 +3,122 @@
 Frequency-dependent foreground components.
 
 This module implements the frequency-dependent component of common foreground
-contaminants. This consist only of factorizable components -- i.e. the components
-in this module describe :math:`f(\nu)` for which the component X has spectrum
-
-.. math:: C_{\ell}^{X} = f(\nu) C^X_{0,\ell}
+contaminants. 
 
 This package draws inspiration from FGBuster (Davide Poletti and Josquin Errard)
 and BeFoRe (David Alonso and Ben Thorne).
 """
+
+from abc import ABC, abstractmethod
 import numpy as np
 from scipy import constants
 
 # TODO: we need to figure out the unit situation.
 
-class FrequencySpectrum:
+T_CMB = 2.725
+
+
+class FrequencySpectrum(ABC):
     """Base class for frequency dependent components."""
 
-    def __init__(self, sed_name=''):
-        """Initialize the component."""
+    @abstractmethod
+    def __call__(self, nu, *args):
+        # Return the evaluation of the SED
         pass
 
-    def __call__(self, nu, **kwargs):
-        """Calls the object's `sed(nu, **kwargs)` method."""
-        return self.sed(nu, **kwargs)
-
 
 class PowerLaw(FrequencySpectrum):
-    r"""
-    FrequencySpectrum for a power law.
-    .. math:: f(\nu) = (nu / nu_0)^{\beta}
+    r""" Power Law
 
-    Parameters
-    ----------
-    nu: float or array
-        Frequency in Hz.
-    beta: float
-        Spectral index.
-    nu0: float
-        Reference frequency in Hz.
-
-    Methods
-    -------
-    __call__(self, nu, beta)
-        return the frequency scaling given by a power law.
+    .. math:: f(\nu) = (\nu / nu_0)^{\beta}
     """
-    def sed(nu, beta, nu_0):
-        """Compute the SED with the given frequency and parameters."""
-        par = self.get_missing(kwargs)
-        return (nu / nu_0)**beta
+
+    def __call__(self, nu, beta, nu0):
```

Comment:
Wow, why did I write it with `sed()` before... This is much clearer.

Reply:
Probably my fault. Much better now